File: snippets/python/src/messages.py
import logging
from breez_sdk_liquid import BindingLiquidSdk, CheckMessageRequest, SignMessageRequest


def sign_message(sdk: BindingLiquidSdk):
    # ANCHOR: sign-message
    message = "<message to sign>"
    try:
        sign_message_response = sdk.sign_message(SignMessageRequest(message))

        # Get the wallet info for your pubkey
        info = sdk.get_info()

        signature = sign_message_response.signature
        pubkey = info.wallet_info.pubkey

        logging.debug(f"Pubkey: {pubkey}")
        logging.debug(f"Signature: {signature}")
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: sign-message

def check_message(sdk: BindingLiquidSdk):
    # ANCHOR: check-message
    message = "<message>"
    pubkey = "<pubkey of signer>"
    signature = "<message signature>"
    try:
        check_message_request = CheckMessageRequest(message, pubkey, signature)
        check_message_response = sdk.check_message(check_message_request)

        is_valid = check_message_response.is_valid

        logging.debug(f"Signature valid: {is_valid}")
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: check-message

File: snippets/go/messages.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-liquid-go/breez_sdk_liquid"
)

func SignMessage(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: sign-message
	message := "<message to sign>"

	signMessageRequest := breez_sdk_liquid.SignMessageRequest{
		Message: message,
	}

	signMessageResponse, err := sdk.SignMessage(signMessageRequest)
	if err != nil {
		log.Printf("Error: %#v", err)
		return
	}

	// Get the wallet info for your pubkey
	info, err := sdk.GetInfo()
	if err != nil {
		log.Printf("Error: %#v", err)
		return
	}

	signature := signMessageResponse.Signature
	pubkey := info.WalletInfo.Pubkey

	log.Printf("Pubkey: %v", pubkey)
	log.Printf("Signature: %v", signature)
	// ANCHOR_END: sign-message
}

func CheckMessage(sdk *breez_sdk_liquid.BindingLiquidSdk) {
	// ANCHOR: check-message
	message := "<message>"
	pubkey := "<pubkey of signer>"
	signature := "<message signature>"

	checkMessageRequest := breez_sdk_liquid.CheckMessageRequest{
		Message:   message,
		Pubkey:    pubkey,
		Signature: signature,
	}

	checkMessageResponse, err := sdk.CheckMessage(checkMessageRequest)
	if err != nil {
		log.Printf("Error: %#v", err)
		return
	}

	isValid := checkMessageResponse.IsValid

	log.Printf("Signature valid: %v", isValid)
	// ANCHOR_END: check-message
}
